src/editor/components/reduce/QueryEditorReduce.tsx

```diff
@@ -74,9 +76,10 @@ export const QueryEditorReduce: React.FC<Props> = props => {
         parameters: parameters,
       };
 
-      props.onChange(payload);
+      onChange(payload); // adding onChange to dependency array below causes maximum call depth error
```

Comment:
I wrote a comment about this one. If I add onChange to dependencies we get a recursion with a maximum call stack depth. Didn't look into this further, but we really should dive deeper into this when we have time.